clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepository.java

```diff
@@ -75,14 +76,24 @@
   private final Set<StoreDataChangedListener> listeners = new CopyOnWriteArraySet<>();
   private final AtomicLong totalStoreReadQuota = new AtomicLong();
 
+  private final long refreshIntervalInSeconds;
+
+  private AtomicBoolean started = new AtomicBoolean(false);
+
   protected NativeMetadataRepository(ClientConfig clientConfig, VeniceProperties backendConfig) {
-    long refreshIntervalInSeconds = backendConfig.getLong(
+    refreshIntervalInSeconds = backendConfig.getLong(
         CLIENT_SYSTEM_STORE_REPOSITORY_REFRESH_INTERVAL_SECONDS,
         NativeMetadataRepository.DEFAULT_REFRESH_INTERVAL_IN_SECONDS);
-    this.scheduler.scheduleAtFixedRate(this::refresh, 0, refreshIntervalInSeconds, TimeUnit.SECONDS);
     this.clientConfig = clientConfig;
   }
 
+  public synchronized void start() {
+    if (!started.get()) {
+      this.scheduler.scheduleAtFixedRate(this::refresh, 0, refreshIntervalInSeconds, TimeUnit.SECONDS);
```

Comment:
Since are shutting down the scheduler in `clear()` I'm wondering if we should also add defensive code to prevent attempts to call `start()` again after calling `clear()`

Reply:
done